src/Raven.Server/Documents/Patch/ScriptRunner.cs

```diff
@@ -861,6 +868,44 @@ private JsValue Raven_Min(JsValue self, JsValue[] args)
                 return args[0];
             }
 
+            private JsValue UnarchiveDoc(JsValue self, JsValue[] args)
+            {
+                if (args.Length != 1)
+                    throw new InvalidOperationException("unarchive(doc) must be called with a single argument");
+                 
+                if (args[0].IsNull() || args[0].IsUndefined())
+                    return args[0];
+                
+                if ((args[0].IsObject() && args[0].AsObject() is BlittableObjectInstance) == false)
+                    throw new InvalidOperationException("unarchive(doc) must take document object as the first argument");
+                
+                var archivedDocId = GetIdFromArg(args[0], _unarchiveSignature);
+                using (var doc = _database.DocumentsStorage.Get(_docsCtx, archivedDocId, DocumentFields.Data, throwOnConflict: true))
+                {
+                    if (doc.Flags.HasFlag(DocumentFlags.Archived) == false)
+                    {
+                        return JsValue.Undefined;
```

Comment:
in general - do we want to return nothing, the same document?
in this particular line - do we want to throw here?

Reply:
returning undefined, throwing errors
 if its not archived, its a noop and we return undefined